templates/code-style-validation.yml

```diff
@@ -23,17 +25,24 @@ jobs:
   # Use Python 3 for validating the code.
   - task: UsePythonVersion@0
     inputs:
-      versionSpec: '3.x'
+      versionSpec: '3.7'
```

Comment:
Let's not get caught by Microsoft supporting 3.8 suddenly.

Reply:
Is this something we might want to make a parameter (that defaults to 3.7)?  Right now, we're going to be targeting 3.7 for all the repos we add azure pipeline CI to, but I could see at some point in the future needing to do another large compatibility pass (Python 4?) and wanting to be able to have different code validation for the PR for that change.

Reply:
Good question. The versionSpec does not say which version of the language we want to target, but simply the version of Python we want to run. If you wanted to target a specific version of the language when validating code, you would configure the tools accordingly, like we do for black.

```
 - repo: https://github.com/ambv/black
    rev: stable
    hooks:
    - id: black
      language_version: python3
```

This tells Black that the Python 3 syntax is considered as valid. If we set it to Python 2 then black would complain about your code if you used `async` for example.